src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -141,66 +142,111 @@ def create_driver(self, input_list, output_list):
                      funcnames=["PSyDataType"])
         prog.add(use)
 
+        # Use a symbol table to make sure all variable names are unique
+        sym_table = GOSymbolTable()
+        sym = Symbol("PSyDataType")
+        sym_table.add(sym)
+
+        psy_data = sym_table.new_symbol_name("psy_data")
+        sym_table.add(Symbol(psy_data))
         var_decl = TypeDeclGen(prog, datatype="PSyDataType",
-                               entity_decls=["psy_data"])
+                               entity_decls=[psy_data])
         prog.add(var_decl)
 
         call = CallGen(prog,
-                       "psy_data%OpenRead(\"{0}\", \"{1}\")"
-                       .format(module_name, region_name))
+                       "{0}%OpenRead(\"{1}\", \"{2}\")"
+                       .format(psy_data, module_name, region_name))
         prog.add(call)
 
         post_suffix = self._post_name
+
+        # Variables might need to be renamed in order to guarantee unique
+        # variable names in the driver: An example of this would be if the
+        # user code contains a variable 'dx', and the kernel takes a
+        # property 'dx' as well. In the original code that is no problem,
+        # since the property is used via field%grid%dx. But the stand-alone
+        # driver renames field%grid%dx to dx, which can cause a name clash.
+        # Similar problems can exist with any user defined type, since all
+        # user defined types are rewritten to just use the field name.
+        # We use a mapping to support renaming of variables: it takes as
```

Comment:
This feels like something that the SymbolTable should be doing? Maybe using the concept of 'tag' that Sergi has recently introduced.

Reply:
I had a look at that, but as far as I understand the 'tag' functionality it seems to be doing something else: I could get a unique name for ``psy_data`` (which might be ``psy_data_27``), and then just use the tag ``psy_data`` to get the actual name. I don't think that helps here, since I an just simply use the symbol directory.

But I would agree that renaming variables in the symbol table would be great. Each symbol when used would use a 'renamed name' or so. But that's a lot of unrelated work (e.g. if we rename 'a' to 'b', then we can't add a symbol 'b' anymore, but we could add another 'a', ...).  The renaming I need here is very limited, and I think an stay the way it is for now.

Reply:
Yes, I think my comment was a bit ambitious! That's something for the future.